### PR TITLE
fix: Adds validation styles to datepicker

### LINKED ISSUE #2263: [fix] DatePicker error handling not up to current USWDS spec
**ReactUSWDS Version & USWDS Version:**
react-uswds: v4.1.0
uswds: v3.1.0

**Describe the bug**
There is a tooltip error that pops up on DatePicker component when there is a wrong input.
This is not compliant to the USWDS current style that highlights the whole component and displays an error message above the input. This should be the same error-handling for the phone number and zipcode inputs as well.

**To Reproduce**

1. Go to a DatePicker field
2. Type in text, anything that is not a proper date.
3. Submit or hit Enter.

**Expected behavior**
The label and input should be highlighted in red with a red error message displayed above input.

**Screenshots**
Current:
image
USWDS compliant: (phone number used as example)
image

**Additional context**

**Device and Browser Information (please complete the following information if describing a UI bug):**

 - Mac Ventura 13.2.1 (22D68)
 - Chrome 
 - Version 110.0.5481.100

### PR DESCRIPTION
**Pull request by werdnanoslen**
# Summary

Adds validation styles (and corresponding story) to DatePicker component, which uses a standard `<input>` instead of our input component, which already has that validation.

## Related Issues or PRs

Resolves #2263 

## How To Test

1. Open http://localhost:9009/?path=/docs/components-date-picker--complete-date-picker
2. Check each `validationStatus` control

### Screenshots (optional)
image

### CODE REVIEW

**Review by brandonlenz — CHANGES_REQUESTED** [review]
Hey [looks familiar](https://github.com/trussworks/react-uswds/pull/2365) 😆 

With a couple unit tests to check that the right class is applied when the prop is set, I'm good to approve

File: src/components/forms/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ type BaseDatePickerProps = {
   id: string
   name: string
   className?: string
+  validationStatus?: 'error' | 'success'
```
Comment (brandonlenz) [inline src/components/forms/DatePicker/DatePicker.tsx:32]: In the PR I linked I made a `ValidationStatus` type that could be reused here as well

Comment (brandonlenz) [reply src/components/forms/DatePicker/DatePicker.tsx:32]: If we merge yours first, I can update it in mine

Comment (werdnanoslen) [reply src/components/forms/DatePicker/DatePicker.tsx:32]: Please do, just pushed the test

File: src/components/forms/DatePicker/DatePicker.test.tsx [hunk 2]
```diff
@@ -21,13 +21,32 @@ describe('DatePicker component', () => {
...
+    }
+    const { getByLabelText, getByTestId, getByText } = render(<DatePicker {...allProps} />)
+    const queryForDatePicker = () => screen.queryByTestId('date-picker')
+    return {
+      getByLabelText,
+      getByTestId,
+      getByText,
+      queryForDatePicker,
+    }
+  }
```
Comment (brandonlenz) [inline src/components/forms/DatePicker/DatePicker.test.tsx:39]:
```suggestion
    const queryForDatePicker = () => screen.queryByTestId('date-picker')
    return {
      ...rendered,
      queryForDatePicker,
    }
```

Comment (brandonlenz) [reply src/components/forms/DatePicker/DatePicker.test.tsx:39]: Just a shorthand to expose all the things from the `render` method

**Review by brandonlenz — APPROVED** [review]
One change I recommend making (a small nit/nice to have)

File: src/components/forms/DatePicker/DatePicker.test.tsx (hunk 2 above)
Comment (werdnanoslen) [reply src/components/forms/DatePicker/DatePicker.test.tsx]: TIL thanks!

**Review by brandonlenz — APPROVED** [review]

**Pull request merged**